peoplehr/types.bal

```diff
@@ -818,3 +818,96 @@ public type HolidayDetail record {|
     string EndDate?;
     *Authentication;
 |};
+
+# Represents additional question
+#
+# + QuestionId - Question ID
+# + Value - Value default value
+public type AdditionalQuestion record {
+    string QuestionId;
+    string Value;
+};
+
+# Internal questions for create new applicant
+#
+public type InternalQuestion record {
+    *AdditionalQuestion;
+};
+
+# Represents Document detail
+#
+# + DocumentName - Document name  
+# + Url - URL for the document
+public type Document record {
+    string DocumentName;
+    string Url;
+};
+
+# Represents new applicant
+#
+# + VacancyReference - Vacancy reference for create new applicant
+# + FirstName - First name for create new applicant 
+# + LastName - Last name for create new applicant
+# + Email - Email for create new applicant 
+# + Gender - Gender for create new applicant 
+# + DateOfBirth - Date of birth for create new applicant
+# + PostCode - Post code for create new applicant
+# + Address - Address for create new applicant
+# + PhoneNumber - PhoneNumber for create new applicant
+# + OtherContactDetails - Other contact details for create new applicant
+# + Source - Source for create new applicant
+# + AdditionalQuestions - Source for create new applicant
+# + InternalQuestions - Internal questions for create new applicant
+# + Documents - Documents for create new applicant
+# + Skills - Skills for create new applicant
+# + RecruitmentCost - Recruitment cost for create new applicant
+# + DateLastContacted - Date last contacted for create new applicant
+public type NewApplicant record {|
+    string VacancyReference?;
+    string FirstName;
+    string LastName;
+    string Email?;
+    string Gender?;
+    string DateOfBirth?;
+    string PostCode?;
+    string Address?;
+    string PhoneNumber?;
+    string OtherContactDetails?;
+    string Source?;
+    AdditionalQuestion[] AdditionalQuestions?;
+    InternalQuestion[] InternalQuestions?;
+    Document[] Documents;
+    string Skills;
+    string RecruitmentCost?;
+    string DateLastContacted?;
+    *Authentication;
+    json...;
+|};
+
+# Represents document detail 
+#
+# + ApplicantId - ApplicantId for upload applicant document  
+# + DocumentName - Document name for upload applicant document(<= 100 characters) 
+# + Description - Description for upload applicant document(<= 256 characters) 
+# + File - File for upload applicant document  
+public type NewDocument record {|
+    string ApplicantId;
+    string DocumentName;
+    string Description;
+    string File;
+    *Authentication;
+|};
+
+# Applicant information for duplicate checking
+#
+# + VacancyReference - Vacancy reference for check duplicate applicant 
+# + FirstName - First name for check duplicate applicant(<= 50 characters) 
+# + LastName - Last name for check duplicate applicant(<= 50 characters) 
+# + Email - Email for check duplicate applicant
+public type ApplicantInformation record {|
+    string VacancyReference;
```

Comment:
Shall we make `VacancyReference ` 3 rd key of the record? We need to give priority to Applicant name.